zips/zip-tbd2.md

```diff
@@ -0,0 +1,107 @@
+# Abstract
+
+This is the first version of the [^zip-tbd1] specification. It’s intended to be a basic implementation that can be used as a first iteration to build off as well as a fallback protocol in the future. As such, there may be certain capabilities or features that are intentionally excluded which can be added in later versions.
+
+# Motivation
+
+In order to define an interoperable version of the specification while allowing for extensibility a base version needs to be defined along with the [^zip-tbd1] specification. This first version is intended to be a simple method to define how messages are encrypted and stored on the [^IPFS] network and then subsequently retrieved by a recipient.
+
+# Terminology
+
+The key words "MUST", "MUST NOT", "SHOULD", "SHOULD NOT", "MAY",
+"RECOMMENDED", "OPTIONAL", and "REQUIRED" in this document are to
+be interpreted as described in BCP 14 [^BCP-14]_ when, and only when,
+they appear in all capitals.
+
+# Specification
+
+## Example
+
+Here's a basic example of the MMP URI for the version 1 specification.
+
+`mmp:01:bafybeihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku?ttl=2024-07-02T14:30:00Z#key=Hy9X_k2mLpQrZtNbVc5hA7sDxEuFoP-iQnWyG4M6OjBv`
+
+## Message Location Encoding Scheme
+
+In this version, the intention is to remain focused on storing messages on the [^IPFS] network so that the protocol can operate in a maximally decentralized fashion. In order to accomplish this the message location in the MMP URI MUST be a CID of either version 0 or version 1. Both versions SHOULD be supported. Since CIDv1’s rely on the multicodec specification which includes multiple different encodings via multibase there’s a possibility that a codec is chosen that cannot be decoded. If this is the case, the message MUST be ignored. An implementation MAY return an error to the recipient if they choose. Additionally, for this reason implementers SHOULD use base64url codec by default.
+
+## Message Encryption
+
+In order to properly end-to-end encrypt the media contents and not need to redefine a new encryption process in detail this specification will reuse commonly implemented and defined cryptographic constructions. In this case, since it’s considered out of scope to support multi-messaging capabilities or extremely large file sizes the secretbox construction from libsodium will be sufficient. This makes it easier to implement this functionality safely without requiring re-implementation of the lower level cryptographic functionality which is important for a default fallback protocol.
+
+## Encryption process
+
+The Message Encryption process MUST utilize the secretbox construction from libsodium with the authentication tag attached. The XSalsa20Poly1305 stream cipher MUST be used also. In order to ensure proper usage both the key and nonce MUST not be reused. For this reason it is RECOMMENDED that implementers leverage the `crypto_secretbox_easy` API in libsodium. 
```

Comment:
Note from ZECC Summit: we do not currently depend on XSalsa20Poly1305 anywhere in the Zcash protocol, and use ChaCha20Poly1305 (in particular the IETF version specified in RFC 7539) for the encrypted memos in which the MMP would be stored. It therefore would be simpler (and have a smaller security surface) to use ChaCha20Poly1305 for MMP. The length limit of 1GB means that any differences in maximum length between XSalsa20Poly1305 and ChaCha20Poly1305 do not matter.

Reply:
Thanks for adding this note @str4d. I'm happy to change this. For the second sentence ("It is RECOMMENDED...") do we have any implementation APIs we might want to point to as a reference?

Also, just confirming that we're good with using RFC 7539 vs XChacha20Poly1305 as defined in [draft-arciszewski-xchacha-03 ](https://datatracker.ietf.org/doc/html/draft-arciszewski-xchacha-03). I'd personally prefer consistency around algorithms for a baseline if possible, so whatever is used already is likely the best.